### ISSUE TITLE
Question on how to use Seq. likelihoods to calibrate

### ISSUE THREAD

**Issue by FairyFali** [parameterlab/apricot#1]
Hi Authors, 

Great work. I am curious about the detailed usage of Seq. likelihoods baseline. Because I think it is an easier-to-use method for white-box models but I can not get the details of how to convert the seq. likelihoods into a calibration score. Could you kindly provide more information about that? I may refer to your work in my project if this point is clear. 

Thank you very much.

**Comment by Kaleidophon** [comment]
Hey, thanks for reaching out!

I hope I am understanding your question correctly, otherwise please feel free to correct me or ask follow-up questions.

In our paper, one of the baselines we use for confidence scores are sequence likelihoods, so simply the (length-normalized) likelihood of a sequence, measured by the product of the individual token probabilities. 

We also tried another baseline which used an idea based on Platt scaling: Given a confidence score p, we try to obtain a more calibrated score q by q = a * p + b, where a and b are single scalars tuned on a calibration set using gradient descent.

Please let me know whether this answers your question!

**Comment by FairyFali** [comment]
The following is an example.  

Sequence: "Hello world"

Token probabilities:  P(Hello) = 0.2, P(world) = 0.1

Length: 2

Product of probabilities: $ 0.2 \times 0.1 = 0.02$

Confidence score is the Length-normalized likelihood: $ 0.02^{1/2} \approx 0.1414 $

Do I understand correctly?

**Comment by Kaleidophon** [comment]
Thank you for providing the example, I can see how the details might cause confusion!

First of all, we would use the conditional token probabilities, so p(world | Hello) (but I assumed you meant that, this just for clarification). 

Secondly, I just looked up the exact way we did it in the code, and it is this line [here](https://github.com/parameterlab/apricot/blob/6aea8510ca0da04a27abbb4b5ba39f727c26c342/src/calibration.py#L164):
Using your example probabilities, we would compute (log 0.2 + log 0.1) / 2 approx -0.8495 and then convert back from log space using exp, so exp(-0.8495) = 0.4276.

Using the geometric mean also seems super sensible though, so I would be curious what you find if you compute the sequence likelihood baseline this way!
